packages/tooltip/src/Tooltip.js

```diff
@@ -1,28 +1,58 @@
 import * as React from "react";
 import PropTypes from "prop-types";
+import values from "object.values";
 
 import { Container, StyledTooltip } from "./styledComponents/Tooltip";
+import alignments from "./constants/alignments";
 
-const Tooltip = props => {
-  const content = props.children || props.text;
-  return (
-    <Container>
-      {content && <StyledTooltip {...props}>{content}</StyledTooltip>}
-    </Container>
-  );
+export const getAutoAlign = bodyWidth => containerRef => align => {
+  if (align !== alignments.AUTO) return align;
+  if (containerRef) {
+    const { x } = containerRef.getBoundingClientRect();
+    if (x > bodyWidth / 2) return alignments.RIGHT;
+  }
+  return alignments.LEFT;
 };
 
+class Tooltip extends React.Component {
+  state = {
+    align: this.props.align
+  };
+  componentDidMount = () => {
+    this.setState({
+      align: getAutoAlign(document.body.clientWidth)(this.containerRef)(
```

Comment:
hum... not sure I get what you're asking.

You think this should be customizable by a prop? If so, then we would have to find other way to get the element's position instead of `getBoundingClientRect`, which will be a little harder.

Or you ask this to understand why this is an argument if it will allways be the same? This is just to make it easier to test, so we wont need to mock `document.body`

Reply:
if we use document.body as the frame to calculate when the tooltip would go "outside of the limits" we are assuming that the limits are always the screen limits.

But we might have cases where we dont want to surpass the limits of a different container, a div with overflow hidden for example.

Reply:
ok, no problems for me releasing this version as well, we can open an issue to add this improvement later